cpu/sam3/include/periph_cpu.h

```diff
@@ -1,35 +1,79 @@
-/*
- * Copyright (C) 2015 Freie Universität Berlin
- *
- * This file is subject to the terms and conditions of the GNU Lesser
- * General Public License v2.1. See the file LICENSE in the top level
- * directory for more details.
- */
-
-/**
- * @ingroup         cpu_sam3x8e
- * @{
- *
- * @file
- * @brief           CPU specific definitions for internal peripheral handling
- *
- * @author          Hauke Petersen <hauke.peterse@fu-berlin.de>
- */
-
-#ifndef PERIPH_CPU_H_
-#define PERIPH_CPU_H_
-
-#include "periph/dev_enums.h"
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-/* nothing defined here so far... */
-
-#ifdef __cplusplus
-}
-#endif
-
-#endif /* PERIPH_CPU_H_ */
-/** @} */
+#ifndef PERIPH_CPU_H_
+#define PERIPH_CPU_H_
+
+#include "periph/dev_enums.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @brief   Overwrite the default gpio_t type definition
+ * @{
+ */
+#define HAVE_GPIO_T
+typedef uint32_t gpio_t;
+/** @} */
+
+/**
+ * @brief   Definition of a fitting UNDEF value
+ */
+#define GPIO_UNDEF          (0xffffffff)
+
+/**
+ * @brief   Define a CPU specific GPIO pin generator macro
+ */
+#define GPIO(x, y)          ((PIOA + (x << 9)) | y)
+#define GPIO2(i)            ((PIOA + (i << 9)))
+/**
+ * @brief   Override values for pull register configuration
+ * @{
+ */
+#define HAVE_GPIO_PP_T
+typedef enum {
+    GPIO_NOPULL = 4,        /**< do not use internal pull resistors */
+    GPIO_PULLUP = 9,        /**< enable internal pull-up resistor */
+    GPIO_PULLDOWN = 8       /**< enable internal pull-down resistor */
+} gpio_pp_t;
+/** @} */
+
+/**
+ * @brief   Override flank configuration values
+ * @{
+ */
+#define HAVE_GPIO_FLANK_T
+typedef enum {
+    GPIO_RISING = 1,        /**< emit interrupt on rising flank */
+    GPIO_FALLING = 2,       /**< emit interrupt on falling flank */
+    GPIO_BOTH = 3           /**< emit interrupt on both flanks */
+} gpio_flank_t;
+/** @} */
+
+/**
+ * @brief   Available ports on the STM32F1 family
+ */
+enum {
+    PORT_A = 0,             /**< port A */
+    PORT_B = 1,             /**< port B */
+    PORT_C = 2,             /**< port C */
+    PORT_D = 3,             /**< port D */
+    
+};
+
+
+/**
+ * @brief declare needed generic SPI functions

```

Comment:
behind every other brief are spaces, not here. please remove the spaces after the other brief's